### PR TITLE
FaultInjectionFix - IncludePrimaryChange

### PR DESCRIPTION
**Pull request by xinlian12**
**Issue**
when defining a FaultInjectionEndpoint, includePrimary only indicate whether primary can be a considered and it may not be part the addresses used.  This behavior will be easily causing confusion and add unnecessary challenges for fault injection testing. 

**Solution**
If includePrimary=true, then primary replica address will always be made sure part of the addresses being returned.

**Other changes:**
Added support for PARTITION_IS_SPLITTING server error type

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-test/src/main/java/com/azure/cosmos/test/implementation/faultinjection/FaultInjectionRuleProcessor.java [hunk 1]
```diff
@@ -351,10 +351,23 @@ private Mono<List<URI>> resolvePhysicalAddresses(
                                         addressEndpoints.isIncludePrimary(),
                                         true)
                                     .flatMapIterable(addresses -> {
+                                        // There are two rules need to happens here:
+                                        // 1. if isIncludePrimary is true, then basically make sure primary replica address will always be returned
+                                        // 2. make sure the same replica addresses will be used across different client instances
                                         return addresses
                                             .stream()
+                                            .sorted((o1, o2) -> {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-test/src/main/java/com/azure/cosmos/test/implementation/faultinjection/FaultInjectionRuleProcessor.java:359]: the sorting is only on primary or not, how will it make sure that ordering is maintained across all client instances?

File: sdk/cosmos/tests.yml [hunk 2]
```diff
@@ -71,7 +71,7 @@ stages:
             - $(sub-config-cosmos-azure-cloud-test-resources)
       MatrixConfigs:
         - Name: Cosmos_faultInjection_live_test
-          Path: sdk/cosmos/integration-matrix.json
+          Path: sdk/cosmos/platform-matrix.json
```
Comment (kushagraThapar) [inline sdk/cosmos/tests.yml:74]: why this change?

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos-test/src/test/java/com/azure/cosmos/test/TestSuiteBase.java [hunk 3]
```diff
@@ -55,6 +56,7 @@
 import static org.mockito.Mockito.doAnswer;
 import static org.mockito.Mockito.spy;
 
+@Listeners({TestNGLogListener.class})
```
[the remark this answers is not in the document]
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-test/src/test/java/com/azure/cosmos/test/TestSuiteBase.java:59]: Just nice to have the extra instrumentations for each method, no real impact

File: sdk/cosmos/tests.yml [hunk 4]
```diff
@@ -83,4 +83,4 @@ stages:
       PreSteps:
         - template: /eng/pipelines/templates/steps/install-reporting-tools.yml
       TestGoals: 'clean verify'
-      TestOptions: '$(ProfileFlag)'
+      TestOptions: '$(ProfileFlag) $(AdditionalArgs)'
```
[the remark this answers is not in the document]
Comment (xinlian12) [reply sdk/cosmos/tests.yml:86]: does not really need in this PR, but it passes any extra parameters, in this case is the client telemetry endpoint. I can revert this

File: sdk/cosmos/azure-cosmos-test/src/test/resources/long-testng.xml [hunk 5]
```diff
@@ -29,7 +29,7 @@
       </run>
     </groups>
     <packages>
-      <package name="com.azure.cosmos.*"/>
+      <package name="com.azure.cosmos.test.*"/>
```
[the remark this answers is not in the document]
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-test/src/test/resources/long-testng.xml:32]: just to change to match the package name

File: sdk/cosmos/tests.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (xinlian12) [reply sdk/cosmos/tests.yml]: reverted
